addons/sale/models/sale_order.py

```diff
@@ -699,17 +698,33 @@ def action_quotation_send(self):
         }
 
     def _find_mail_template(self, force_confirmation_template=False):
-        template_id = False
+        """ Get the appropriate mail template for the current sale's order(s) based on their state.
 
-        if force_confirmation_template or (self.state == 'sale' and not self.env.context.get('proforma', False)):
-            template_id = int(self.env['ir.config_parameter'].sudo().get_param('sale.default_confirmation_template'))
-            template_id = self.env['mail.template'].search([('id', '=', template_id)]).id
-            if not template_id:
-                template_id = self.env['ir.model.data']._xmlid_to_res_id('sale.mail_template_sale_confirmation', raise_if_not_found=False)
-        if not template_id:
-            template_id = self.env['ir.model.data']._xmlid_to_res_id('sale.email_template_edi_sale', raise_if_not_found=False)
+        If all SOs are 'sale' or ``force_confirmation_template`` is `True`, the mail template is
+        that of the sale confirmation. Else, it returns the quotation's email template.
 
-        return template_id
+        :param bool force_confirmation_template: Allows to force the confirmation template to be
+        use in any case
+        :return: The correct mail template based on the current status
+        :rtype: record of `mail.template` or `None` if not found
+        """
+        return_template = self.env.ref('sale.email_template_edi_sale', False)
```

Comment:
could be moved in an else clause, not used if we enter the if.
(or another if to fallback when `mail_template_sale_confirmation` is not found, to keep old behavior)